data-serving/data-service/src/controllers/preprocessor.ts

```diff
@@ -259,6 +262,62 @@ export const createCaseRevision = async (
     next();
 };
 
+export const createBatchDeleteCaseRevisions = async (
+    request: Request,
+    response: Response,
+    next: NextFunction,
+): Promise<void> => {
+    let casesToDelete: any;
+    if (request.body.caseIds !== undefined) {
+        casesToDelete = (
+            await Case.find({
+                _id: {
+                    $in: request.body.caseIds,
+                },
+            }).exec()
+        ).map((c) => {
+            return {
+                case: c,
+            };
+        });
+    } else {
+        // If we surpass the maxThreshold, skip creating revisions as we won't
```

Comment:
Good idea, moved to first middleware. 

Interesting. We kind of already have that because in the UI we know how many are going to be deleted. We could probably change the confirmation dialog to an error dialog if the user selects more than 10,000 cases to be deleted. I'll look into that.

Reply:
Added error dialog in #989